src/main/java/io/github/genomicdatainfrastructure/daam/mappers/RemsApplicationMapper.java

```diff
@@ -6,157 +6,241 @@
+                .roles(application.getApplicationRoles())
+                .createdAt(application.getApplicationCreated())
+                .modifiedAt(application.getApplicationModified())
+                .workflow(toWorkflow(application))
+                .applicant(toApplicant(application))
+                .members(toMembers(application))
+                .invitedMembers(toInvitedMembers(application))
+                .datasets(toDatasets(application))
+                .forms(toForms(application))
+                .permissions(toPermissions(application))
+                .events(toEvents(application))
+                .attachments(toAttachments(application))
+                .licenses(toLicences(application))
+                .state(toState(application))
                 .build();
     }
 
-    private static RetrievedApplicationWorkflow toRetrievedApplicationWorkflow(
-            Response10953Workflow remsWorkflow) {
-        return new RetrievedApplicationWorkflow(remsWorkflow.getWorkflowId(),
-                remsWorkflow.getWorkflowType());
+    private RetrievedApplicationWorkflow toWorkflow(Application application) {
+        Optional<Response10953Workflow> workflow = Optional.ofNullable(application
+                .getApplicationWorkflow());
+
+        return new RetrievedApplicationWorkflow(workflow.map(Response10953Workflow::getWorkflowId)
+                .orElse(null),
+                workflow.map(Response10953Workflow::getWorkflowType).orElse(null));
     }
 
-    private static RetrievedApplicationApplicant toRetrievedApplicationApplicant(
-            UserWithAttributes remsApplicant) {
+    private RetrievedApplicationApplicant toApplicant(Application application) {
+        Optional<UserWithAttributes> applicant = Optional.ofNullable(application
+                .getApplicationApplicant());
+
         return new RetrievedApplicationApplicant(
-                remsApplicant.getUserid(),
-                remsApplicant.getName(),
-                remsApplicant.getEmail());
+                applicant.map(UserWithAttributes::getUserid).orElse(null),
+                applicant.map(UserWithAttributes::getName).orElse(null),
+                applicant.map(UserWithAttributes::getEmail).orElse(null));
     }
 
-    private static List<RetrievedApplicationMember> toRetrievedApplicationMembers(
-            Set<UserWithAttributes> remsMembers) {
-        return remsMembers
+    private List<RetrievedApplicationMember> toMembers(Application application) {
+        Optional<Set<UserWithAttributes>> potentialMembers = Optional.ofNullable(application
+                .getApplicationMembers());
+
+        return potentialMembers.map(members -> members
                 .stream()
-                .map(member -> new RetrievedApplicationMember(member.getUserid(),
-                        member.getName(),
-                        member.getEmail()))
-                .toList();
+                .map(member -> {
```

Comment:
important: to improve readability, you should assign this lambda function into a anther private method, the same for the other cases.